### PR TITLE
Integration of Similarity into Main flow

### PR DESCRIPTION
**Pull request by karan-modh**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -119,24 +129,53 @@ def main():
...
-
-	print("Following are the new inputs that you entered and were not in the graph. Please add the necessary edges (Deductions/Inferences) for the input")
-	print(not_in_graph_inputs)
 
+	not_in_graph_inputs = list(not_in_graph_inputs)
+	nodes_added = {}
+	data = get_all_objects()
+	objects = data["objects"]
+	object_names = [preprocess_name(element["name"]) for element in objects]
+	for node in not_in_graph_inputs:
```
Comment (Itisha2987) [inline main.py]: Please make functions... It becomes a bit hard to understand in a flow.

File: main.py [hunk 2]
```diff
@@ -119,24 +129,53 @@ def main():
...
+			if not similar_objects:
+				continue
+			most_similar_object = similar_objects[0]
+			most_similar_object_name = preprocess_name(most_similar_object[1]["name"])
+			graph.add_node(node)
+			successors = graph.successors(most_similar_object_name)
+			edge_attributes = {}
+			for successor in successors:
+				graph.add_edge(node, successor)
+				edge_attributes[(node, successor)] = graph.get_edge_data(most_similar_object_name, successor)
```
Comment (Itisha2987) [inline main.py]: Which edges are being added to new input?

File: main.py [hunk 3]
```diff
@@ -119,24 +129,53 @@ def main():
...
+			nx.set_edge_attributes(graph, edge_attributes)
+			nodes_added[node] = (most_similar_object_name, most_similar_object[0])
```
Comment (Itisha2987) [inline main.py]: What is this line doing?

File: main.py (hunk 2 above)
Comment (karan-modh) [reply main.py]: all the outgoing edges of the old object.

File: main.py (hunk 3 above)
Comment (karan-modh) [reply main.py]: Line no. 174?
It is storing new nodes added and the similar object used to add edges and its similarity index

File: main.py (hunk 2 above)
Comment (Itisha2987) [reply main.py]: Outgoing edges contains inferences and deductions also.
Let's say deduction edges which are representing the object properties can't be added blindly. Tomato_soup deduces tangy_smell, but you can't say manchow_soup deduces tangy smell.
So we need to check which deduction edges can be added.

**Pull request merged**